R/scales-.R

```diff
@@ -61,13 +61,44 @@ ScalesList <- ggproto("ScalesList", NULL,
     scale[[1]]
   },
 
+  train = function(self, data, drop = FALSE) {
+    if (length(self$scales) == 0) {
+      return()
+    }
+    lapply(data, self$train_df, drop = drop)
+  },
+
   train_df = function(self, df, drop = FALSE) {
     if (empty(df) || length(self$scales) == 0) {
       return()
     }
     lapply(self$scales, function(scale) scale$train_df(df = df))
   },
 
+  map = function(self, data, layers) {
```

Comment:
I'm not really a fan of this as layers and scales should ideally be separate on only interact through the data, but I haven't found a better way around this.